src/config/SourcegraphConfigOptions.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+final class PhabricatorSourcegraphConfigOptions extends PhabricatorApplicationConfigOptions
+{
+    public function getName()
+    {
+        return pht('Sourcegraph');
+    }
+
+    public function getDescription()
+    {
+        return pht('Configure Sourcegraph Settings.');
+    }
+
+    public function getIcon()
+    {
+        return 'fa-asterisk';
+    }
+
+    public function getGroup()
+    {
+        return 'apps';
+    }
+
+    public function getOptions()
+    {
+        $repo_type = 'sourcegraph.repo';
+        $repo_example = array(
+            array(
+                'path' => 'gorilla/mux',
+                'callsign' => 'MUX',
+            ),
+        );
+
+        $repo_example = id(new PhutilJSON())->encodeAsList(
+            $repo_example);
+
+        return array(
+            $this->newOption(
+                'sourcegraph.url',
+                'string',
+                null)
+                ->setDescription(pht('URL to Sourcegraph.'))
+                ->addExample('https://sourcegraph.example.com', pht('Valid Setting')),
+            $this->newOption('sourcegraph.repos', $repo_type, array())
+                ->setDescription(pht(
+                    'This option is only required for repositories defined in `%s` array found in your Sourcegraph site admin configuration.' .
+                    "\n\n" .
+                    'The object is an array with all elements of the type object. The array object has the following properties:' .
```

Comment:
The first sentence is not helpful to me. "What is the type object? Which type? Is the first usage of the word 'object' referring to the same thing as the second usage in the sentence?"

Instead:

```
An array of objects, each mapping a Phabricator repository's callsign to the corresponding repository on Sourcegraph. Each object contains the following properties:
```